File: examples/remote-types/tests/bindings/test_remote_types.swift
import remote_types

let testLogger = try! LogSink(path: "SomeFile")
testLogger.log(level: LogLevel.info, msg: "Hello world")

// Test the error handling
do {
    let _ = try LogSink(path: "")
    fatalError("Constructor should have thrown")
} catch is AnyhowError {
    // Expected
}

File: examples/remote-types/tests/bindings/test_remote_types.rb
# frozen_string_literal: true

require 'test/unit'
require 'remote_types'

class TestRemoteTypes < Test::Unit::TestCase
  include RemoteTypes

  def test_logger
    testLogger = LogSink.new 'SomeFile'
    testLogger.log LogLevel::INFO, 'Hello world'
  end

  def test_error_handling
    assert_raises AnyhowError do
      LogSink.new ''
      raise RuntimeError, 'Constructor should have thrown'
    end
  end
end
